backend/src/main/kotlin/uk/gov/justice/digital/backend/service/AthenaPreviewService.kt

```diff
@@ -0,0 +1,46 @@
+package uk.gov.justice.digital.backend.service
+
+import jakarta.inject.Singleton
+import uk.gov.justice.digital.backend.client.preview.PreviewClient
+import uk.gov.justice.digital.backend.converter.DomainToPreviewQueryConverter
+import uk.gov.justice.digital.backend.repository.DomainRepository
+import uk.gov.justice.digital.model.Status
+import kotlin.math.min
+
+interface PreviewService {
+    fun preview(domainName: String, status: Status, limit: Int): List<Map<String, String>>
+
+    companion object {
+        const val MaximumLimit = 100
+    }
+}
+
+@Singleton
+class AthenaPreviewService(private val client: PreviewClient,
+                           private val repository: DomainRepository,
+                           private val converter: DomainToPreviewQueryConverter): PreviewService {
+
+    override fun preview(domainName: String, status: Status, limit: Int): List<Map<String, String>> {
+        val domains = repository.getDomains(domainName, status)
+
+        val domain = domains.firstOrNull()
+            ?: throw DomainNotFoundException("No domains found for name: $domainName status: $status")
+
+        // TODO - only domains with a single table are supported
```

Comment:
Is there a ticket to cover this? Or is it included in a future feature?

Reply:
There should be something on the backlog for this.